packages/aws-cdk-lib/aws-dynamodb/lib/global-table.ts

```diff
@@ -0,0 +1,2152 @@
+
+/**
+ * Options for configuring a system errors metric that considers multiple operations.
+ */
+export interface SystemErrorsForOperationsMetricOptions extends cloudwatch.MetricOptions {
+
+    /**
+     * The operations to apply the metric to.
+     *
+     * @default - All operations available by DynamoDB tables will be considered.
+     */
+    readonly operations?: Operation[];
+
+}
+
+/**
+ * Options for configuring metrics that considers multiple operations.
+ */
+export interface OperationsMetricOptions extends SystemErrorsForOperationsMetricOptions { }
+
+/**
+ * Supported DynamoDB table operations.
+ */
+export enum Operation {
+
+    /** GetItem */
+    GET_ITEM = 'GetItem',
+
+    /** BatchGetItem */
+    BATCH_GET_ITEM = 'BatchGetItem',
+
+    /** Scan */
+    SCAN = 'Scan',
+
+    /** Query */
+    QUERY = 'Query',
+
+    /** GetRecords */
+    GET_RECORDS = 'GetRecords',
+
+    /** PutItem */
+    PUT_ITEM = 'PutItem',
+
+    /** DeleteItem */
+    DELETE_ITEM = 'DeleteItem',
+
+    /** UpdateItem */
+    UPDATE_ITEM = 'UpdateItem',
+
+    /** BatchWriteItem */
+    BATCH_WRITE_ITEM = 'BatchWriteItem',
+
+    /** TransactWriteItems */
+    TRANSACT_WRITE_ITEMS = 'TransactWriteItems',
+
+    /** TransactGetItems */
+    TRANSACT_GET_ITEMS = 'TransactGetItems',
+
+    /** ExecuteTransaction */
+    EXECUTE_TRANSACTION = 'ExecuteTransaction',
+
+    /** BatchExecuteStatement */
+    BATCH_EXECUTE_STATEMENT = 'BatchExecuteStatement',
+
+    /** ExecuteStatement */
+    EXECUTE_STATEMENT = 'ExecuteStatement',
+
+}
+
+/**
+ * Represents an attribute for describing the key schema for the table
+ * and indexes.
+ */
+export interface Attribute {
+    /**
+     * The name of an attribute.
+     */
+    readonly name: string;
+
+    /**
+     * The data type of an attribute.
+     */
+    readonly type: AttributeType;
+}
+
+/**
+ * What kind of server-side encryption to apply to this table.
+ */
+export enum TableEncryption {
+    /**
+     * Server-side KMS encryption with a master key owned by AWS.
+     */
+    DEFAULT = 'AWS_OWNED',
+
+    /**
+     * Server-side KMS encryption with a customer master key managed by customer.
+     * If `encryptionKey` is specified, this key will be used, otherwise, one will be defined.
+     *
+     * > **NOTE**: if `encryptionKey` is not specified and the `Table` construct creates
+     * > a KMS key for you, the key will be created with default permissions. If you are using
+     * > CDKv2, these permissions will be sufficient to enable the key for use with DynamoDB tables.
+     * > If you are using CDKv1, make sure the feature flag `@aws-cdk/aws-kms:defaultKeyPolicies`
+     * > is set to `true` in your `cdk.json`.
+     */
+    CUSTOMER_MANAGED = 'CUSTOMER_MANAGED',
+
+    /**
+     * Server-side KMS encryption with a master key managed by AWS.
+     */
+    AWS_MANAGED = 'AWS_MANAGED',
+}
+
+/**
+ * Represents the table schema attributes.
+ */
+export interface SchemaOptions {
```

Comment:
I don't like this type. Remember how the existence of this type, shared as it was everywhere, made it impossible for us to make a change we needed to make to address one issue? We talked about this.

I hope you're not using it below :). If you are, just copy/paste these two fields into where you were using it before.